developer_manual/app/tutorial.rst

```diff
@@ -0,0 +1,487 @@
+
+        public function __construct(IDb $db) {
+            parent::__construct($db, 'ownnotes_notes', '\OCA\OwnNotes\Db\Note');
+        }
+
+        public function find($id, $userId) {
+            $sql = 'SELECT * FROM *PREFIX*ownnotes_notes WHERE id = ? AND user_id = ?';
+            return $this->findEntity($sql, [$id, $userId]);
+        }
+
+        public function findAll($userId) {
+            $sql = 'SELECT * FROM *PREFIX*ownnotes_notes WHERE user_id = ?';
+            return $this->findEntities($sql, [$userId]);
+        }
+
+    }
+
+.. note:: The first parent constructor is the database layer, the second one the table that is used and the third is the entity on which the result should be mapped onto. Insert, delete and update methods are already implemented.
+
+Connect Database & Controllers
+==============================
+The mapper which provides the database access is finished and can be passed into the controller.
+
+In general it is good practice to use another class between mappers and controllers called services, but for the sake of brevity we will leave them out.
+
+You can pass in the mapper by adding it as a type hinted parameter. ownCloud will figure out how to :doc:`assemble them by itself <container>`. Additionally we want to know the userId of the currently logged in user. Simply add a **$UserId** parameter to the constructor (case sensitive!). To do that open **ownnotes/controller/notecontroller.php** and change it to the following:
+
+.. code-block:: php
+
+   <?php
+    namespace OCA\OwnNotes\Controller;
+
+    use \Exception;
+
+    use \OCP\IRequest;
+    use \OCP\AppFramework\Http\Http;
+    use \OCP\AppFramework\Http\DataResponse;
+
+    use \OCA\OwnNotes\Db\Note;
+    use \OCA\OwnNotes\Db\NoteMapper;
+
+    class NoteController extends \OCP\AppFramework\Controller {
+
+        private $mapper;
+        private $userId;
+
+        public function __construct($AppName, IRequest $request, NoteMapper $mapper, $UserId){
+            parent::__construct($AppName, $request);
+            $this->mapper = $mapper;
+            $this->userId = $UserId;
+        }
+
+        /**
+         * @NoAdminRequired
+         */
+        public function index() {
+            return new DataResponse($this->mapper->findAll($this->userId));
+        }
+
+        /**
+         * @param int $id
+         * @NoAdminRequired
+         */
+        public function show($id) {
+            try {
+                return new DataResponse($this->mapper->find($id, $this->userId));
+            } catch(Exception) {
+                return new DataResponse([], Http::STATUS_NOT_FOUND);
+            }
+        }
+
+        /**
+         * @NoAdminRequired
+         */
+        public function create($title, $content) {
+            $note = new Note();
+            $note->setTitle($title);
+            $note->setContent($content);
+            $note->setUserId($this->userId);
+            return $this->mapper->insert($note);
```

Comment:
Is it intended that the return values in `update()` for example are within a `DataResponse` and here we return just the `insert`?